builder/builder-next/worker/worker.go

```diff
@@ -49,8 +49,16 @@ import (
 	"golang.org/x/sync/semaphore"
 )
 
+var (
+	// buildkitVersion holds the version number. Filled in at linking time.
+	buildkitVersion = "v0.0.0+unknown"
+	// buildkitRevision holds the VCS revision. Filled in at linking time.
+	buildkitRevision = ""
+)
```

Comment:
It's really a shame we can't get this directly from `go.mod` (even via something like https://pkg.go.dev/runtime/debug#ReadBuildInfo, but it could be more direct if it were a first-order language feature).

Maybe we could do something like abusing `go:embed` to embed the `go.mod` itself and re-parse it? :confused:

(Overall, I think we should trend more towards a raw `go build` being "enough" for the most part, so anything that requires movement in the other direction should be weighed carefully IMO)

Would love to hear more maintainer opinions :bow:

Reply:
> (Overall, I think we should trend more towards a raw `go build` being "enough" for the most part, so anything that requires movement in the other direction should be weighed carefully IMO)

Yeah I also don't really like having this logic in the build pipeline but this information is still useful for analytics and debugging purpose when someone reports an issue imo. If only go tooling could provide git information for a dependency that would be handy.

Reply:
Is there not a way to get this from `ReadBuildInfo`?

I would have thought `BuildInfo.Deps[].Version` would have the information we want? I know we can't actually use this easily in BuildKit itself, since there's no way to actually get the version for the main module (it's left empty in `BuildInfo.Main`), but I think we should be able to use it here fairly easily?